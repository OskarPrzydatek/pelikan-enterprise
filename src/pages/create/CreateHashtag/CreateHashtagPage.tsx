import { SubmitHandler, useForm } from 'react-hook-form';

import { CreateHashtagTemplate } from '~/components/templates';

interface IHashtag {
  hashtagName: string;
}

export const CreateHashtagPage: React.FC = () => {
  const methods = useForm<IHashtag>();

  // eslint-disable-next-line no-console
  const onSubmit: SubmitHandler<IHashtag> = (data) => console.log(data);

  return (
    <CreateHashtagTemplate
      creatHashtagFormMethods={methods}
      onSubmitCreateHashtag={methods.handleSubmit(onSubmit)}
    />
  );
};
