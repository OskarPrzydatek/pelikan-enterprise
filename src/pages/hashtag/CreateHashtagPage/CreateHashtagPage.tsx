import { SubmitHandler, useForm } from 'react-hook-form';

import { CreateHashtagTemplate } from '~/components/templates';
import { IHashtag } from '~/models';

export const CreateHashtagPage: React.FC = () => {
  const methods = useForm<IHashtag>();

  // eslint-disable-next-line no-console
  const onSubmit: SubmitHandler<IHashtag> = (data) => console.log(data);

  return (
    <CreateHashtagTemplate
      creatHashtagFormMethods={methods}
      onSubmitCreateHashtag={methods.handleSubmit(onSubmit)}
    />
  );
};
