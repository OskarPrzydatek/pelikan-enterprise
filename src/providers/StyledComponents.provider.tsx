import { ThemeProvider } from 'styled-components';

import { GlobalStyleLayout } from '../styles/layouts';
import { theme } from '../styles/theme';

interface IStyledComponentsProvider {
  children: React.ReactNode;
}

export const StyledComponentsProvider: React.FC<IStyledComponentsProvider> = ({
  children,
}: IStyledComponentsProvider) => {
  return (
    <ThemeProvider theme={theme}>
      <GlobalStyleLayout />
      {children}
    </ThemeProvider>
  );
};
