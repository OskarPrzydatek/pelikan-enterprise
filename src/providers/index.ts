export { StyledComponentsProvider } from './StyledComponents.provider';
