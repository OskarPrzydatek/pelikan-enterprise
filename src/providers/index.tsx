import { MemoryRouter, Route, Routes } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';

import { GlobalStyleLayout } from '../styles/layouts';
import { theme } from '../styles/theme';

interface IStyledComponentsProvider {
  children: React.ReactNode;
}

interface ITestingPageProvider {
  element: JSX.Element;
}

export const StyledComponentsProvider: React.FC<IStyledComponentsProvider> = ({
  children,
}: IStyledComponentsProvider) => {
  return (
    <ThemeProvider theme={theme}>
      <GlobalStyleLayout />
      {children}
    </ThemeProvider>
  );
};

export const TestingPageProvider: React.FC<ITestingPageProvider> = ({
  element,
}: ITestingPageProvider) => {
  return (
    <StyledComponentsProvider>
      <MemoryRouter initialEntries={['/page1']}>
        <Routes>
          <Route element={element} path="/:page" />
        </Routes>
      </MemoryRouter>
    </StyledComponentsProvider>
  );
};
