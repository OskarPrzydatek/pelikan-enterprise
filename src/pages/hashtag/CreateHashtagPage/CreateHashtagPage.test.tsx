import { render } from '@testing-library/react';
import { describe } from 'vitest';

import { TestingPageProvider } from '~/providers';

import { CreateHashtagPage } from './CreateHashtagPage';

const MockCreateHashtagPage: React.FC = () => (
  <TestingPageProvider element={<CreateHashtagPage />} />
);

describe('CreateHashtagPage', () => {
  test('page snapshot', () => {
    const view = render(<MockCreateHashtagPage />);
    expect(view).toMatchSnapshot();
  });
});
