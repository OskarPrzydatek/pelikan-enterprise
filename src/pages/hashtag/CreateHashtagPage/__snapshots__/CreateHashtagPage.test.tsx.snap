// Vitest Snapshot v1

exports[`CreateHashtagPage > page snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-gswNZR cJrdyV"
      >
        <form
          class="sc-jIRcFI iwFYYu"
        >
          <h2
            class="sc-eDvSVe dcXtK"
          >
            Dodaj hashtag
          </h2>
          <div
            class="sc-dmctIk iThEeJ"
          >
            <div
              class="sc-kgTSHT fmvgAY"
            >
              <input
                class="sc-fLcnxK gGYQKp"
                data-testid="create-hashtag-name"
                id="name"
                name="name"
                placeholder="Nazwa hashtagu"
              />
            </div>
          </div>
          <input
            class="sc-hhOBVt dvAfkB"
            type="submit"
            value="Dodaj"
          />
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-gswNZR cJrdyV"
    >
      <form
        class="sc-jIRcFI iwFYYu"
      >
        <h2
          class="sc-eDvSVe dcXtK"
        >
          Dodaj hashtag
        </h2>
        <div
          class="sc-dmctIk iThEeJ"
        >
          <div
            class="sc-kgTSHT fmvgAY"
          >
            <input
              class="sc-fLcnxK gGYQKp"
              data-testid="create-hashtag-name"
              id="name"
              name="name"
              placeholder="Nazwa hashtagu"
            />
          </div>
        </div>
        <input
          class="sc-hhOBVt dvAfkB"
          type="submit"
          value="Dodaj"
        />
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
